### PR TITLE
Allow Invensense sensors to be used at 2kHz backend rate on faster processors

### PR DESCRIPTION
**Pull request by andyp1per**
On my small 3" copter I am right on the Nyquist frequency at half throttle (500Hz for a backend sampling rate of 1Khz). The backend rate is used as the sample rate by the low pass filters and notch filters. For the low pass filter passing the Nyquist may not be a problem, but for the harmonic notch tracking the motor rpm it will not be able to go above the Nyquist frequency.

The simple fix for faster processors and IMUs supporting fast sampling is to set the default backend rate to 2Khz rather than 1Khz

### CODE REVIEW

**Comment by andyp1per** [comment]
Here is a KakuteF7Mini flying with this:

image

and the FFT to show it really is running at 2Khz

image

works well and is such an obvious change I can't believe I didn't think of it before!

@lthall FYI

**Comment by andyp1per** [comment]
Flew hard on this today, going past the Nyquist frequency for a 1Khz backend rate. All was fine, but I think this is pretty critical enhancement for smaller copters:

image

I also flew this on an F4 (Pixracer) with FFT and double-notch enabled. No particular issues, so this could be made an option for F4 processors as well.

File: libraries/AP_InertialSensor/AP_InertialSensor_Invensense.cpp [hunk 1]
```diff
@@ -55,6 +55,13 @@ extern const AP_HAL::HAL& hal;
 
 #define MPU_SAMPLE_SIZE 14
 #define MPU_FIFO_BUFFER_LEN 16
+#if defined(STM32H7) || defined(STM32F7)
```
Comment (tridge) [inline libraries/AP_InertialSensor/AP_InertialSensor_Invensense.cpp:58]: this is a nice idea, just please add a comment explaining why this is being done

**Comment by chobitsfan** [comment]
Thank you @andyp1per 

I happen to see this PR. This is my test flight log using this PR with KakuteF7 and Tekko32 4in1 esc
[esc_telem.zip](https://github.com/ArduPilot/ardupilot/files/4655244/esc_telem.zip)

File: libraries/AP_InertialSensor/AP_InertialSensor_Invensense.cpp (hunk 1 above)
Comment (andyp1per) [reply libraries/AP_InertialSensor/AP_InertialSensor_Invensense.cpp:65]: Comment added

**Comment by andyp1per** [comment]
I think this should be closed in favour of #14441

**Comment by rmackay9** [comment]
@andyp1per,

I'll close this one if that's OK.

**Pull request closed without merging**